### PR TITLE
Navbar improvements: dropdown support, margin/padding variables for navbar items, calculated spacer for body for fixed navbars

### PR DESCRIPTION
**Pull request by Mevrael**
Fixes: #17532: Dropdowns in navbar navs? from #18467
1. Padding in navbar removed to allow bg color on :hover on nav-item. Margin in nav-item now can be used instead.
2. Added padding and margin to navbar nav-items configurable via 4 new variables:
   - $navbar-item-padding-y: 0.5rem;
   - $navbar-item-padding-x: 1rem;
   - $navbar-item-margin-y: 0;
   - $navbar-item-margin-x: 0;
3. Fixed navbar-brand to position correctly
4. Added dropdown support in navbar.
5. Added example navbar-dropdown with bg-inverse, right align and fixed navbar.
6. Added two new classes for body to add calculated margin if fixed navbar used:
   - `.body-navbar-top-spacer`
   - `.body-navbar-bottom-spacer`

### CODE REVIEW

File: docs/examples/navbar-dropdown/index.html [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            <a class="nav-link" href="#">About us</a>
+        </li>
+    </ul>
+
+    <ul class="nav navbar-nav pull-xs-right">
+        <li class="nav-item">
+            <a class="nav-link" href="#">Admin panel</a>
+        </li>
+        <li class="nav-item dropdown">
+            <a class="nav-link dropdown-toggle" id="nav_user_dropdown" data-toggle="dropdown" href="#" role="button" aria-haspopup="true" aria-expanded="false">
```
Comment (cvrebert) [inline docs/examples/navbar-dropdown/index.html]: Example IDs should use camelCase instead of underscore_separated_words

**Comment by Mevrael** [comment]
@cvrebert changed id to camelCase. If you will give me link to that part in style guide because I could not found id naming convention in it, would be perfect. I've used Google style guides and hadn't seen camelCase ids so much.

**Comment by cvrebert** [comment]
It's not presently codified anywhere to my knowledge, though you can see it being adhered to in e.g. [the forms docs](http://v4-alpha.getbootstrap.com/components/forms/). I don't think we advocate that convention in real-world code, but it arguably makes them more distinct from classes in the examples.

**Comment by Mevrael** [comment]
Primarily as a server side sr. softwrae engineer I would recommend for IDs and especially form input names using snake_case and for classes/URLs lisp-case but never camelCase. While for IDs camelCase is often used for form inputs there is only one option.

Form input names, for example, in backend usually represents variable, array key or database column name and snake_case is used there for purpose. It is also much readable. Form input names also might be an array and I've never seen camelCase[arrays],. camelCase commonly used in object-oriented programming languages and should not be used in HTML/CSS at all.

And using snake_case for IDs also avoids having three naming conventions in one document. Once I've written corporate standards where I clearly defined that IDs and input form names should be snake_cased.

**Comment by Mevrael** [comment]
@mdo any comments on this almost one month old PR?

**Comment by mdo** [comment]
@Mevrael Hey, sorry about the lack of update on some of these PRs, yours included. Been swamped and trying to focus on the bigger (next alpha) rewrites before getting to some of this other stuff.

**Comment by bottleofwater** [comment]
This `.body-navbar-top-spacer` utility seems quite useful.

**Comment by mdo** [comment]
Closing as stale based on the navbar overhauling we did in the releases published since this PR was opened originally. Major PR was #19890.

**Pull request closed without merging**